backend/app/usecase/match/keyword.go

```diff
@@ -0,0 +1,32 @@
+package match
+
+import (
+	"fmt"
+)
+
+// KeywordType represents keyword matching type.
+type KeywordType int
```

Comment:
Actually, it is not Keyword type. It is a type of matching algorithm. Can we name this something else to make it obvious?

Reply:
How about changing to Algorithm?

Reply:
Sounds good to me. Yeah, `match.Algorithm` sounds better than `match.KeywordType`